File: Part8/Fortran/FunctionConcepts/FunctionConcepts_2.f90
! An example of function variable scope
program main
    integer :: Global_Variable = 15
    integer :: Global_To_Be_Shadowed = 5
    
    print *, "Global variable:", Global_Variable
    print *, "Global shadowed:", Global_To_Be_Shadowed
    call my_function ()
    print *, "Function variable:", My_Variable
    print *, "Global variable:", Global_Variable
    print *, "Global shadowed:", Global_To_Be_Shadowed
    
contains
    subroutine my_function ()
        integer :: My_Variable = 55
        integer :: Global_To_Be_Shadowed = 15
        Global_Variable = 42
    end subroutine my_function
end program main

File: Part8/C#/FunctionConcepts/FunctionConcepts_2.cs
// An example of function variable scope
using System;

class Program
{
    static int GlobalVariable = 15;
    static int GlobalToBeShadowed = 5;

    static void MyFunction ()
    {
        int myVariable = 55;
        GlobalVariable = 42;
        int GlobalToBeShadowed = 15;
    }

    static void Main ()
    {
        Console.WriteLine ( "Global variable: {0}", GlobalVariable );
        Console.WriteLine ( "Global shadowed: {0}", GlobalToBeShadowed );
        MyFunction ();
        Console.WriteLine ( "Function variable: {0}", myVariable );
        Console.WriteLine ( "Global variable: {0}", GlobalVariable );
        Console.WriteLine ( "Global shadowed: {0}", GlobalToBeShadowed );
    }
}
